portfolio/src/main/webapp/tasksApi.js

```diff
@@ -0,0 +1,203 @@
+var GoogleAuth;
+
+// Scope for read access to Tasks API
+const SCOPE_TASKS_READ = 'https://www.googleapis.com/auth/tasks.readonly';
+
+const TASKS_CLIENT_ID =
+    '499747085593-hvi6n4kdrbbfvcuo1c9a9tu9oaf62cr2.apps.googleusercontent.com';
+
+const TASKS_DISCOVERY_DOCS =
+    'https://www.googleapis.com/discovery/v1/apis/tasks/v1/rest';
+
+var TASKS_API_KEY = '';
+
+function fetchApiKey() {
+  fetch('./appConfigServlet')
```

Comment:
It seems like multiple javascript files might be fetching the same app config. Consider implementing a file like appConfig.js that caches the app config. That way files like tasksApi.js can call appConfig.js, which may return the cached config.

Reply:
In a soon to come pull request, we will only be calling for the config one time.